### PR TITLE
Updated KUBECTL_VERSION to 1.31.0 for fixing vulnerabilities

### PR DESCRIPTION
**Pull request by jdesouza**
```
usr/local/bin/kubectl (gobinary)

Total: 3 (HIGH: 2, CRITICAL: 1)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                           Title                            │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24790 │ CRITICAL │ fixed  │ 1.21.7            │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for │
│         │                │          │        │                   │                 │ IPv4-mapped IPv6 addresses                                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790                 │
│         ├────────────────┼──────────┤        │                   ├─────────────────┼────────────────────────────────────────────────────────────┤
│         │ CVE-2023-45288 │ HIGH     │        │                   │ 1.21.9, 1.22.2  │ golang: net/http, x/net/http2: unlimited number of         │
│         │                │          │        │                   │                 │ CONTINUATION frames causes DoS                             │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2023-45288                 │
│         ├────────────────┤          │        │                   ├─────────────────┼────────────────────────────────────────────────────────────┤
│         │ CVE-2024-34156 │          │        │                   │ 1.22.7, 1.23.1  │ encoding/gob: golang: Calling Decoder.Decode on a message  │
│         │                │          │        │                   │                 │ which contains deeply nested structures...                 │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-34156                 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────────────────────┘
```

### CODE REVIEW

**Comment by mozillazg** [comment]
@jdesouza Thanks for your contribution! It looks like this issue was fixed via https://github.com/aquasecurity/kube-bench/pull/1687

**Comment by jdesouza** [comment]
Looks good although a kubectl vulnerability is likely to remain

**Comment by mozillazg** [comment]
> Looks good although a kubectl vulnerability is likely to remain

@jdesouza Could you please update this PR to upgrade only the kubectl version? Thanks!

**Comment by jdesouza** [comment]
> > Looks good although a kubectl vulnerability is likely to remain
> 
> @jdesouza Could you please update this PR to upgrade only the kubectl version? Thanks!

WDYT now?

**Review by afdesk — APPROVED** [review]
LGTM! Thanks! 

@mozillazg wdyt?

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/aquasecurity/kube-bench
 
-go 1.22
+go 1.22.7
```
[the remark this answers is not in the document]
Comment (afdesk) [reply go.mod:3]: @mozillazg I definitely agree with you!
my point was that we already use Go 1.22.7 for building in pipelines.

**Comment by afdesk** [comment]
Alpine version was alredy bumped via #1676

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdesouza) [reply go.mod]: Thanks for the review! Just reverted the go change. Now we have just the kubectl update

**Comment by afdesk** [comment]
@jdesouza @mozillazg thanks for your efforts!

**Pull request merged**